csharp/src/Vista.SDK/Internal/LocalIdError.cs

```diff
@@ -0,0 +1,62 @@
+namespace Vista.SDK.Internal;
+
+public sealed record LocalIdError(IEnumerable<(ParsingState, string)> Errors)
+{
+    public static implicit operator LocalIdError((ParsingState type, string message) error) =>
+        new LocalIdError(new[] { (error.type, error.message) });
+}
+
+public sealed record LocalIdErrorBuilder
+{
+    private readonly List<(ParsingState type, string message)> _errors;
+    private static Dictionary<ParsingState, string> _predefinedErrorMessages =>
+        SetPredefinedMessages();
+
+    public static readonly LocalIdErrorBuilder Empty = new LocalIdErrorBuilder();
+
+    internal LocalIdErrorBuilder() => _errors = new List<(ParsingState, string)>();
+
+    internal LocalIdErrorBuilder AddError(ParsingState state, string? message)
+    {
+        if (string.IsNullOrWhiteSpace(message))
+            return AddError(state);
+
+        _errors.Add((state, message!));
```

Comment:
It's possible to do `LocalIdErrorBuilder.Empty.AddError(...)` right now, which feels dangerous. Any way to avoid that?